codalab/rest/bundles.py

```diff
@@ -1314,7 +1315,12 @@ def delete_bundles(uuids, force, recursive, data_only, dry_run):
             local.model.delete_bundles(relevant_uuids)
 
         # Update user statistics
-        local.model.update_user_disk_used(request.user.user_id)
+        # Just decrement the user disk used by the sum of sizes of bundles deleted
+        # OK, now let's add our change.
+        bundle_data_sizes = local.model.get_bundle_metadata(relevant_uuids, 'data_size')
```

Comment:
Does this diff produce the same answer as the above?  Just wondering if there are subtle differences?

Reply:
It did not, it turns out, due to the following edge cases:
- cl rm -d: Here, the bundle contents are removed but not the bundle metadata. For a given bundle B, we must take care to decrement the user disk quota when *cl rm -d B* is run, but to not decrement it again when *cl rm B* is run (since that second *cl rm* just deletes the metadata and not anything from disk).
- Files uploaded with symlinks: These shouldn't affect user disk quota regardless, so when we remove them there should be no decrement to the user disk quota.

The new changes fix this.